### DISCUSSION TITLE
git cloning vs. copying directory and best practices for git cloning.

### DISCUSSION THREAD

**Discussion by dmastrop** [community/community#53148] [Programming Help] [answered]
### Body

hi all

I have some very basic questions about git cloning and why it would be an advantage to do this over copying the directory 

I have copied the directory of a project and it seems to have worked well, tested the code and deployment and both the "forks" seem to be working fine independently (removed the remote name and created a new remote git repo for it and then remapped the remote name to the new remote git repo with a new name)

What is the advantage of cloning?. I have never cloned a local repo before. 
I looked at the commit history of the copied folder vs. the original and all of the commit history was present.

Are there some best practices on doing this to avoid complications?

Once the remote repo is cloned to a new local folder is there anything that one has to do to in terms of cleanup?
**_I assume the remote name has to be removed and remapped to a new remote git repository that I should create and that's about it._**
Then do a push to the new remote repo within the new folder to populate the new remote repo for this clone.

I did read the section 2.1 in the book that discusses this.
In this syntax
git clone https://github.com/libgit2/libgit2 mylibgit

Is this the url of the repository in the browser?
Reason why I ask is that I see 
https://github.com/username/my_remote_repo_name
whereas example above has .../libgit2/libgit2   

Also is fork the older term that means basically the same thing as clone?

warm regards

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by dmastrop** [comment]
I did a practice run on a test repo and see that the remote name of "origin" is used, but that this maps to the original remote repo.   Is there a reason why they would map this to the original repo instead of just deleting the remote repo configuration on the clone, and letting the user create a new remote for the clone? (as best practice)

warm regards

**Comment by airtower-luna** [comment, marked as answer]
> What is the advantage of cloning?. I have never cloned a local repo before.

Most likely there isn't one. There are two differences:

* Copying will also copy repository-internal config (stuff in `.git/`).
* Copying will also copy and untracked/modified files in the working tree.

Both of those may be desirable or not. :slightly_smiling_face: 

> In this syntax
> git clone https://github.com/libgit2/libgit2 mylibgit
> 
> Is this the url of the repository in the browser?

The first parameter you give to `git clone` is the location of the repository you want to clone. It can be any kind of URL Git understands, or the directory path of a local repository. That location becomes the location of the default `origin` remote.

> Also is fork the older term that means basically the same thing as clone?

No. Fork can mean different things, all of them different from "clone":

* In general, a fork of a software means that someone took existing (usually open source) code, and continued development in their own direction, more or less independent of the original software. This often happens because people disagree about the direction development should take, or if development on the original software stopped and someone else picked it up on their terms.
* On GitHub (and many other Git hosting platforms) a fork is a copy of another repository that's linked to the original ("upstream") repository, allowing the owner of the fork repository to make PRs to suggest their changes to the upstream developers. See also: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/about-forks

**Reply by dmastrop** [reply]
Thanks Airtower. On the test repo that I cloned the only difference I noticed was that the development (second branch) was not cloned over.  And that would be ok.  With the copy I noticed all the branches were copied over.  

Many thanks for the link.
